#162948853 Get read time for articles

kwameasiago opened the pull request:
**What does this PR do?**
Creates a view for getting the reading time for an article
**Description of tasks to be completed**

- Create a view for getting the reading time of an article
- Created test for reading time

**How should this be manually tested?**

- git clone https://github.com/andela/ah-django-unchained.git
- git checkout ft-article-rating-162948843
- source .env
- python migrate
- run tests python manage.py test

On POSTMAN register and login in a user via http://127.0.0.1:8000/api/users/
create an article via http://127.0.0.1:8000/api/articles/
get the read time via http://127.0.0.1:8000/api/articles/read/`slug`/

**Relevant Pivotal Tracker Story**
[#162948853](https://www.pivotaltracker.com/story/show/162948853)

kwameasiago committed:
feat (read-time) Get read time for articles
 - Create a view for read time of an article
- Create test for read time view

[Delivers #162948853]

d-kahara commented on authors/apps/articles/views.py:
```diff
@@ -56,3 +57,25 @@ class DeleteArticle(UpdateAPIView):
+            return Response({'error': 'article was not found'}, status.HTTP_404_NOT_FOUND)
+        serializer = self.serializer_class(article_instance)
+        words = serializer.data['body'].split()
+        read = len(words) / 279
```
Consider adding a comment indicating that `279` is the average number of words a person reads per minute

kwameasiago committed:
Feat (read-time) Add comment for getting readtime
- Add a comment to explain the how the read time is calculated

[Delivers #162948853]

kwameasiago committed:
--amend

ignatiusukwuoma commented on authors/apps/articles/tests/test_create_article.py:
```diff
@@ -202,7 +237,34 @@ def test_unauthorized_delete_article(self):
+        self.post_article_articles(self.small_body)
+        read_url = reverse('articles:articles-read', kwargs={'slug': 'small'})
+        response = self.client.get(read_url, format='json')
+        data = {'read_time': 'Less than a minute read minute read'}
```
typo

ignatiusukwuoma commented on authors/apps/articles/views.py:
```diff
@@ -131,3 +132,25 @@ def put(self, request, slug):
                                            context={'request': request},
                                            partial=True)
         return Response(serializer.data, status.HTTP_200_OK)
+class ReadTime(RetrieveAPIView):
```
Leave two spaces before class

ignatiusukwuoma commented on authors/apps/articles/views.py:
```diff
@@ -131,3 +132,25 @@ def put(self, request, slug):
+            read = 'Less than a minute read'
+        else:
+            read = round(read)
+        return Response({'read_time': '{} minute read'.format(read)}, status.HTTP_200_OK)
```
Just return the amount of time (in minutes)... Let the frontend handle how it is displayed

ignatiusukwuoma commented on authors/apps/articles/urls.py:
```diff
@@ -14,5 +15,8 @@
     path('articles/<slug>/dislike/',
-         DislikeArticleApiView.as_view(), name='dislikes')
+         DislikeArticleApiView.as_view(), name='dislikes'),
+    path('articles/read/<slug:slug>/',
```
What's the difference between `<slug:slug>` and `<slug>`

kwameasiago replied:
I had used <slug: slug> because the detail view was using <slug:  slug> but since the output is still the same I switched to <slug>

kwameasiago committed:
feat (read time) Fix typo
- Add two lines after class
- Change <slug: slug> to <slug>

[Delivers #162948853]

ignatiusukwuoma commented on authors/apps/articles/views.py:
```diff
@@ -131,3 +132,27 @@ def put(self, request, slug):
+        I picked 279 as an estimate readtime """
+        read = len(words) / 279
+        if read < 1:
+            read = 'Less than a minute read'
```
You should return the same datatype in any case. How about returning `round(read)` in any case, the frontend can sort how to display.

kwameasiago committed:
feat (read time) Return read time
- Return the read time only

[Delivers #162948853]

Georgeygigz commented on authors/apps/articles/views.py:
```diff
@@ -4,9 +4,10 @@
 from django.core.exceptions import ObjectDoesNotExist
 from django.core import serializers
 from rest_framework import status
+from django.core.exceptions import ObjectDoesNotExist
```
Remove repeated import

kwameasiago committed:
--amend

kwameasiago committed:
--amend

kwameasiago committed:
--amend
